lib/model/model.go

```diff
@@ -253,24 +255,44 @@ func NewModel(cfg config.Wrapper, id protocol.DeviceID, clientName, clientVersio
 }
 
 func (m *model) serve(ctx context.Context) error {
-	// Add and start folders
+	defer m.closeAllConnectionsAndWait()
+
+	if err := m.initFolders(); err != nil {
+		close(m.started)
+		return fatalErr(err)
+	}
+
+	m.cfg.Subscribe(m)
+	defer m.cfg.Unsubscribe(m)
+	close(m.started)
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-m.fatalChan:
+		return err
```

Comment:
Seeing as we're never going to return a non-fatal error we could do the wrapping just the once in the return here and have the channel just take normal errors from anywhere